rclcpp/src/rclcpp/duration.cpp

```diff
@@ -182,6 +182,36 @@ Duration::operator-(const rclcpp::Duration & rhs) const
     rcl_duration_.nanoseconds - rhs.rcl_duration_.nanoseconds);
 }
 
+void
+bounds_check_duration_scale(int64_t dns, double scale, uint64_t max)
+{
+  auto abs_dns = (uint64_t)std::abs(dns);
+  auto abs_scale = (uint64_t)std::abs(scale);
+
+  if ((abs_scale >= 1.0 && abs_dns > (uint64_t)(max / abs_scale)) ||
+    (abs_scale < 1.0 && (uint64_t)(abs_dns * abs_scale > max)))
```

Comment:
If we're below an absolute scaling factor of 1 the value can only get smaller which will never overflow or underflow a signed integer. I think this whole 2nd case can be dropped.